### ISSUE TITLE
always generates update and version subcommands

### ISSUE THREAD

**Issue by robertgzr** [hofstadter-io/hofmod-cli#8]
I'm explicitly setting the schema fields to `false`: 

```
#CLI: schema.#Cli & {
	Name: "test"
	Package: "example.com/test/test/cmd/test"
	Short: "a test"

	CompletionCommands: false
	Updates: false
	
    OmitRun: false

	Pflags: [...schema.#Flag] & [{
		Name: "test"
		Long: "test"
		Short: "t"
		Type: "string"
		Default: ""
		Help: "test test test"
	}]

	// Commands: []
}
```

and while they are not added to the `Cmd` in `root.go` the actual commands still get generated:
```
cmd/test
├── cmd
│  ├── completions.go
│  ├── hls
│  │  └── cli
│  │     └── root
│  │        └── help.hls
│  ├── root.go
│  ├── root_test.go
│  ├── update.go
│  └── version.go
├── flags
│  └── root.go
├── main.go
└── verinfo
   └── verinfo.go
```

which makes the build fail like this:
```
building go files...
go build -o test ./cmd/test
go: finding module for package example.com/test/test/cmd/test/ga
cmd/test/cmd/completions.go:8:2: cannot find module providing package example.com/test/test/cmd/test/ga: module example.com/test/test/cmd/test/ga: git ls-remote -q origin in /home/user/go/pkg/mod/cache/vcs/BLAAAAAAAAAA: exit status 128:
	ERROR: Repository not found.
	fatal: Could not read from remote repository.
	
	Please make sure you have the correct access rights
	and the repository exists.
make: *** [Makefile:10: build] Error 1
```

**Comment by verdverm** [comment]
The issue appears to be that this field should be an optional field in the schema, rather than a bool.

The check is here: https://github.com/hofstadter-io/hofmod-cli/blob/master/gen/cli.cue#L111

Try updating your local schema to use the optional (`Updates?: bool`) and then removing it from your design.

**Comment by robertgzr** [comment]
@verdverm cool that helps with excluding update commnand. my issue remains, I still get version and completions since they seem to have the same problem

making them optional as well fixes this but I feel like that's rather an issue with the checks in the file you linked, no?

```
if In.CLI.Updates != _|_ {
   // foo
}
```

doesn't evaluate to `false` when I'm passing `false`... I don't understand CUE well enough but shouldn't it be checking for `In.CLI.Updates != false` there

**Comment by robertgzr** [comment]
another thing to note...

```
go: finding module for package example.com/test/test/cmd/test/ga
```

is the root of the problem for me. I don't have that package and it's not generated if it should be... I would appreciate a switch to exclude google analytics entirely since I would actually want to use update-, version- and completion commands

**Comment by robertgzr** [comment]
```patch
From 01c7bc8eebb929c92e82fe7e95d3292d6d0fe94d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert=20G=C3=BCnzler?= <r@gnzler.io>
Date: Fri, 23 Oct 2020 19:09:02 +0200
Subject: [PATCH] templates: remove leftover references to ga.SendCommand

should be dropped when `Telemetry` is not set
---
 templates/completions.go | 2 ++
 templates/update.go      | 4 ++++
 templates/version.go     | 2 ++
 3 files changed, 8 insertions(+)

diff --git a/templates/completions.go b/templates/completions.go
index df365bc..bc6bf88 100644
--- a/templates/completions.go
+++ b/templates/completions.go
@@ -5,7 +5,9 @@ import (
 
 	"github.com/spf13/cobra"
 
+	{{ if .CLI.Telemetry }}
 	"{{ .CLI.Package }}/ga"
+	{{ end }}
 )
 
 var (
diff --git a/templates/update.go b/templates/update.go
index 6a3c621..223d486 100644
--- a/templates/update.go
+++ b/templates/update.go
@@ -14,7 +14,9 @@ import (
   "github.com/spf13/cobra"
 	"github.com/parnurzeal/gorequest"
 
+	{{ if .CLI.Telemetry }}
 	"{{ .CLI.Package }}/ga"
+	{{end}}
 	"{{ .CLI.Package }}/verinfo"
 )
 
@@ -55,7 +57,9 @@ var UpdateCmd = &cobra.Command{
 	Long: UpdateLong,
 
 	PreRun: func(cmd *cobra.Command, args []string) {
+		{{ if .CLI.Telemetry }}
 		ga.SendCommandPath("update")
+		{{ end }}
 
 	},
 
diff --git a/templates/version.go b/templates/version.go
index 831b1d3..4c6f736 100644
--- a/templates/version.go
+++ b/templates/version.go
@@ -7,7 +7,9 @@ import (
 
 	"github.com/spf13/cobra"
 
+	{{ if .CLI.Telemetry }}
 	"{{ .CLI.Package }}/ga"
+	{{end}}
 	"{{ .CLI.Package }}/verinfo"
 )
 
-- 
2.26.2
```

**Comment by robertgzr** [comment]
the above worked for me :)

**Comment by verdverm** [comment]
Nice @robertgzr! I actually knew about that ga one and wanted to see if you could figure it out, to see how we are doing on our end. I have those fixes and another one or two locally.

Also adding the ability to embed static files if that would be useful for you.

**Comment by verdverm** [comment]
fixed in `v0.7.0`, keeping the original schema rather than the suggested change
